pkg/clusteragent/autoscaling/workload/controller.go

```diff
@@ -224,23 +231,21 @@ func (c *Controller) syncPodAutoscaler(ctx context.Context, key, ns, name string
 			}
 
 			podAutoscalerInternal.Generation = podAutoscaler.Generation
-			c.store.Set(key, podAutoscalerInternal, c.ID)
 		}
 	}
 
 	// Implement sync logic for local ownership, source of truth is Kubernetes
 	if podAutoscalerInternal.Spec.Owner == datadoghq.DatadogPodAutoscalerLocalOwner {
 		if podAutoscalerInternal.Generation != podAutoscaler.Generation {
 			podAutoscalerInternal.UpdateFromPodAutoscaler(podAutoscaler)
-			c.store.Set(key, podAutoscalerInternal, c.ID)
 		}
 	}
 
+	// Reaching this point, we had an error in processing, clearing up global error
+	podAutoscalerInternal.Error = nil
+
 	// Now that everything is synced, we can perform the actual processing
 	result, err := c.handleScaling(ctx, podAutoscaler, &podAutoscalerInternal)
```

Comment:
Should we handle this error ?

Reply:
We're clearing up internal state error to see if reconcile is going to populate it again.

Reply:
I meant the return of `result, err := c.handleScaling`

Reply:
It's handled lines 251:257